### PR TITLE
Add Evil theme for ADD, EDIT and DELETE buttons in Editor 

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR adds conversion of first 19 images in CELLWIN ICN to the Evil interface that corresponds to some buttons used in Editor and checkbox backgrounds.

img

### CODE REVIEW

**Review by ihhub — APPROVED** [review]

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -3228,6 +3228,23 @@ namespace fheroes2
...
+
+                if ( _icnVsSprite[ICN::CELLWIN].size() > 18 ) {
+                    // Convert to Evil only the first 19 images. The rest are not standard buttons and are player color related settings used in original editor.
+                    _icnVsSprite[ICN::CELLWIN_EVIL].resize( 19 );
+                    std::copy( _icnVsSprite[ICN::CELLWIN].begin(), _icnVsSprite[ICN::CELLWIN].begin() + 19, _icnVsSprite[ICN::CELLWIN_EVIL].begin() );
+
+                    for ( Sprite & outputImage : _icnVsSprite[ICN::CELLWIN_EVIL] ) {
+                        // To convert the yellow borders of some items the combination of good-to-evil and gray palettes is used here.
+                        fheroes2::ApplyPalette( outputImage, 0, 0, outputImage, 0, 0, outputImage.width(), outputImage.height(),
+                                                PAL::CombinePalettes( PAL::GetPalette( PAL::PaletteType::GOOD_TO_EVIL_INTERFACE ),
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:3242]: We can cache `PAL::CombinePalettes() output outside this loop as this is not an instant operation.

File: src/fheroes2/dialog/dialog_selectitems.cpp [hunk 2]
```diff
@@ -1486,10 +1482,7 @@ void Dialog::selectMineType( int32_t & type, int32_t & color )
     std::array<fheroes2::Rect, resourceCount> resourceRoi;
 
     // Resource type selection mark background.
-    fheroes2::Sprite markBackground = fheroes2::AGG::GetICN( ICN::CELLWIN, 4 );
-    if ( isEvilInterface ) {
-        fheroes2::ApplyPalette( markBackground, PAL::CombinePalettes( PAL::GetPalette( PAL::PaletteType::GRAY ), PAL::GetPalette( PAL::PaletteType::DARKENING ) ) );
-    }
+    const fheroes2::Sprite markBackground = fheroes2::AGG::GetICN( isEvilInterface ? ICN::CELLWIN_EVIL : ICN::CELLWIN, 4 );
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_selectitems.cpp:1485]: This variable should be reference.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I took a second look and left 2 comments here. Would you mind to check them in your free time?

File: src/fheroes2/dialog/dialog_selectitems.cpp (hunk 2 above)
Comment (Districh-ru) [reply src/fheroes2/dialog/dialog_selectitems.cpp]: Thanks, @ihhub, for noticing this!

File: src/fheroes2/agg/agg_image.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/agg/agg_image.cpp]: Done!

**Review by ihhub — APPROVED** [review]

**Pull request merged**